runners/flink/runner/src/main/java/org/apache/beam/runners/flink/translation/FlinkStreamingTransformTranslators.java

```diff
@@ -172,19 +205,21 @@ public void translateNode(Read.Unbounded<T> transform, FlinkStreamingTranslation
 
       DataStream<WindowedValue<T>> source;
       if (transform.getSource().getClass().equals(UnboundedFlinkSource.class)) {
-        UnboundedFlinkSource flinkSource = (UnboundedFlinkSource) transform.getSource();
+        @SuppressWarnings("unchecked")
+        UnboundedFlinkSource<T> flinkSource = (UnboundedFlinkSource<T>) transform.getSource();
         source = context.getExecutionEnvironment()
             .addSource(flinkSource.getFlinkSource())
-            .flatMap(new FlatMapFunction<String, WindowedValue<String>>() {
+            .flatMap(new FlatMapFunction<T, WindowedValue<T>>() {
               @Override
-              public void flatMap(String s, Collector<WindowedValue<String>> collector) throws Exception {
-                collector.collect(WindowedValue.<String>of(s, Instant.now(), GlobalWindow.INSTANCE, PaneInfo.NO_FIRING));
+              public void flatMap(T s, Collector<WindowedValue<T>> collector) throws Exception {
+                collector.collect(WindowedValue.of(s, Instant.now(), GlobalWindow.INSTANCE, PaneInfo.NO_FIRING));
               }
-            }).assignTimestampsAndWatermarks(new IngestionTimeExtractor());
+            }).assignTimestampsAndWatermarks(new IngestionTimeExtractor<WindowedValue<T>>());
```

Comment:
`SourceFunction`s don't have to assign timestamps or emit watermarks when we are using Event Time. In Flink, you can always assign them later. In the Beam Runner we default to Event Time which makes it tricky for us because the user can't actually extract watermarks/timestamps via the Flink API. Thus, we defaulted to Ingestion Time here to give users a good out of the box experience, e.g. when they read data from Kafka. We are currently reworking some of the provided sources to emit timestamps/watermarks directly in the source.

Note that, this limitation applies only to Flink wrapped sources. The general wrapper for unbounded sources emits watermarks and timestamped values.